---
title: "Sam_GO_test_Cerebellum_data"
author: "Devika Agarwal"
date: 'Last update: `r date()`'
output:
  html_document:
    number_sections: true
    toc: true
    toc_float:
      collapsed: false
      smooth_scroll: yes
      toc_depth: 6
---
<style type="text/css">

body{ /* Normal  */
      font-size: 12px;
  }
td {  /* Table  */
  font-size: 10px;
}
h1.title {
  font-size: 20px;
  color: DarkRed;
}
h1 { /* Header 1 */
  font-size: 20px;
  color: DarkBlue;
}
h2 { /* Header 2 */
    font-size: 18px;
  color: DarkBlue;
}
h3 { /* Header 3 */
  font-size: 14px;
  font-family: "Times New Roman", Times, serif;
  color: DarkBlue;
}
code.r{ /* Code block */
    font-size: 12px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 14px;
}
</style>
```{r setup, include=FALSE, message=FALSE, warning=FALSE}
library(Seurat)
library(dplyr)
library(Matrix)
library(data.table)
library(ggplot2)
library(biomaRt)
library(rhdf5)
library(clusterProfiler)
library(AnnotationDbi)
library(org.Hs.eg.db)
knitr::opts_chunk$set(cache=TRUE,echo=T,message=FALSE,warning=FALSE,cache.lazy = FALSE)
## white background theme for all plots
theme_set(theme_bw())

```


# Load Sams seurat object 
+ Get all genes in the SCT object, as SCT object used for differential expression
+ Filter out genes not expressed in any cells
+ Filter out genes not expressed (>1) in atleast 3 cells and use the final list for background genes for GO Ontology enrichment

```{r }
Sams_seurat <- readRDS(file="./20190807vs2_v3update_newganoidscombined.rds")
SCT_obj <- GetAssayData(object=Sams_seurat, slot="counts", assay="SCT")
# Remove genes not expressed in any cells
cat("Number of genes and cells in the SCT object:", dim(SCT_obj), fill=T)
cat("Number of genes with no UMI counts in any cell: ", sum(rowSums(SCT_obj) == 0), fill=T)
cat("Number of genes with exactly a count of 1 in a single cell (genes with row sum of 1):", sum(rowSums(SCT_obj) == 1), fill=T)

# We define a gene as detectable if at least 3 cells contain more than 1 transcript from the gene
keep_feature <- rowSums(SCT_obj >= 1) >3 
SCT_obj <- SCT_obj[keep_feature, ]

cat("Number of genes and samples after filtering of genes based on UMI counts >= 1 in atleast 3 cells : ", dim(SCT_obj), fill=T)

# use this list for background genes for  GO term over enrichment analysis with ClusterProfiler

allgenes <- rownames(SCT_obj)
```


# Load DEG lists at population level first
```{r pressure, echo=FALSE}
pop0 <- read.table(file="./population_level/20190906DEPop0MGvsCTRL.csv", sep=",", header=T)
pop1 <- read.table(file="./population_level/20190906DEPop01MGvsCTRL.csv", sep=",", header=T)
pop2 <- read.table(file="./population_level/20190906DEPop02MGvsCTRL.csv", sep=",", header=T)
pop3 <- read.table(file="./population_level/20190906DEPop03MGvsCTRL.csv", sep=",", header=T)
pop4 <- read.table(file="./population_level/20190906DEPop04MGvsCTRL.csv", sep=",", header=T)
pop5 <- read.table(file="./population_level/20190906DEPop05MGvsCTRL.csv", sep=",", header=T)
pop6 <- read.table(file="./population_level/20190906DEPop06MGvsCTRL.csv", sep=",", header=T)
pop7 <- read.table(file="./population_level/20190906DEPop07MGvsCTRL.csv", sep=",", header=T)
pop8 <- read.table(file="./population_level/20190906DEPop08MGvsCTRL.csv", sep=",", header=T)
pop9 <- read.table(file="./population_level/20190906DEPop09MGvsCTRL.csv", sep=",", header=T)
pop10 <- read.table(file="./population_level/20190906DEPop10MGvsCTRL.csv", sep=",", header=T)
pop11 <- read.table(file="./population_level/20190906DEPop11MGvsCTRL.csv", sep=",", header=T)

# Filter the DEG list to only include significant genes based on p_val <0.05
pop0_s <- (pop0 %>% filter(p_val <0.05) %>% dplyr::select(.,X))
pop1_s <- pop1 %>% filter(p_val <0.05) %>% dplyr::select(.,X)
pop2_s <- pop2 %>% filter(p_val <0.05) %>% dplyr::select(.,X)
pop3_s <- pop3 %>% filter(p_val <0.05) %>% dplyr::select(.,X)
pop4_s <- pop4 %>% filter(p_val <0.05) %>% dplyr::select(.,X)
pop5_s <- pop5 %>% filter(p_val <0.05) %>% dplyr::select(.,X)
pop6_s <- pop6 %>% filter(p_val <0.05) %>% dplyr::select(.,X)
pop7_s <- pop7 %>% filter(p_val <0.05) %>% dplyr::select(.,X)
pop8_s <- pop8 %>% filter(p_val <0.05) %>% dplyr::select(.,X)
pop9_s <- pop9 %>% filter(p_val <0.05) %>% dplyr::select(.,X)
pop10_s <- pop10 %>% filter(p_val <0.05) %>% dplyr::select(.,X)
pop11_s <- pop11 %>% filter(p_val <0.05) %>% dplyr::select(.,X)

# As clusterProfile CompareCluster requires a list of list , we make a list of lists with the shortened population genes
pop_DEG <- list(pop0=as.character(pop0_s$X),pop1=as.character(pop1_s$X),pop2= as.character(pop2_s$X),pop3= as.character(pop3_s$X),pop4= as.character(pop4_s$X),pop5=as.character(pop5_s$X),pop6= as.character(pop6_s$X),pop7= as.character(pop7_s$X) ,pop8= as.character(pop8_s$X),pop9= as.character(pop9_s$X),pop10= as.character(pop10_s$X),pop11=as.character(pop11_s$X))
```

# Population level DEG GO terms Enrichment
+ Not all populations showed enrichments for GO terms

```{r}
# background gene Universe
background_universe <- as.character(rownames(SCT_obj))

population_DEG_GO <- compareCluster(pop_DEG, universe = background_universe, pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05, maxGSSize = 1000, keyType="SYMBOL", OrgDb= org.Hs.eg.db, readable=F, fun="enrichGO", ont="BP")

# Remove Redundant GO terms

population_DEG_GO_simplified <- simplify(population_DEG_GO, cutoff=0.7)
population_DEG_GO_df <- as.data.frame(population_DEG_GO)
population_DEG_GO_simplified_df <- as.data.frame(population_DEG_GO_simplified)
```

# Dotplot: Top enriched go terms per cluster
+ Plot with all terms 

```{r,fig.height=8, fig.width=9}
dotplot(population_DEG_GO, showCategory = 10) + theme(axis.text.y = element_text(size=8), axis.text.x = element_text(size=7)) +scale_color_viridis_c(direction = 1)
```


+ Plot  after removing redundant terms

```{r,fig.height=8, fig.width=9}
dotplot(population_DEG_GO_simplified, showCategory = 10) + theme(axis.text.y = element_text(size=8), axis.text.x = element_text(size=7)) +scale_color_viridis_c(direction = 1)
```

# MG vs Control DEG genes

+ 3 stimulated samples vs 3 control samples

```{r}
DEG_stim <- read.table(file="./DE_stim20190905.csv", sep=",", header=T)

DEG_stim_sig <- subset(DEG_stim, DEG_stim$p_val < 0.05, select=1)

DEG_stim <- as.character(DEG_stim_sig$X)
```

```{r}
DEG_stim_GO <- enrichGO(DEG_stim, universe = background_universe, pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05, maxGSSize = 1000, keyType="SYMBOL", OrgDb= org.Hs.eg.db, readable=F, ont="BP")
DEG_stim_GO_simplified <- simplify(DEG_stim_GO, cutoff=0.7)
DEG_stim_GO_df <- as.data.frame(DEG_stim_GO)
DEG_stim_GO_simplified_df <- as.data.frame(DEG_stim_GO)
```

+ Plot of all terms enriched in stimulation
+ No redundant terms were removed

```{r}
barplot(DEG_stim_GO_simplified, showCategory = 10) + theme(axis.text.y = element_text(size=8), axis.text.x = element_text(size=7)) +scale_fill_viridis_c(direction = 1)
```
```{r}
write.table(population_DEG_GO,file="./population_stimulated_DEG_GO_enrichment.txt", sep="\t", quote=F)
write.table(population_DEG_GO_simplified, file="./population_stimulated_DEG_GO_enrichment_noredundant.txt", sep="\t",quote=F)
write.table(DEG_stim_GO, file="./DEG_Treatment_GO_terms.txt", sep="\t", quote=F)
```

